AWSTranscribeStreamingUnitTests/AWSTranscribeStreamingClientWebSocketProviderTests.swift

```diff
@@ -0,0 +1,332 @@
+        mockWebSocketProvider.disconnect()
+        wait(for: [receiveClosed], timeout: 10)
+    }
+    
+    // Given: A web socket provider
+    // When: when registering the streaming client
+    // Then: the streaming client is able to use a third party web socket provider and open the socket and send data
+    func testWSSSendData() {
+        let mockWebSocketProvider = MockWebSocketProvider()
+        mockWebSocketProvider.setErrorOnConnectionStatusDidChange(error: nil)
+        guard let config = AWSServiceConfiguration(
+            region: .USEast1,
+            credentialsProvider: MockReturningCredentialsProvider()) else {
+                XCTFail("Couldn't initialize config with MockErrorReturningCredentialsProvider")
+                return
+        }
+        
+        AWSTranscribeStreaming.register(with: config, forKey: "testWSSInitWithWebSocketProvider", webSocketProvider: mockWebSocketProvider)
+        
+        let client = AWSTranscribeStreaming(forKey: "testWSSInitWithWebSocketProvider")
+        
+        guard let request = AWSTranscribeStreamingStartStreamTranscriptionRequest() else {
+            XCTFail("request unexpectedly nil")
+            return
+        }
+        let transcription = "some transcription"
+        let data = Data(transcription.utf8)
+        
+        request.languageCode = .enUS
+        request.mediaEncoding = .pcm
+        request.mediaSampleRateHertz = 8000
+        
+        let delegate = MockTranscribeStreamingClientDelegate()
+        
+        let receiveDataSend = expectation(description: "Callback received data")
+        delegate.connectionStatusCallback = { status, error in
+            
+            if status == AWSTranscribeStreamingClientConnectionStatus.connected {
+                //send data once socket is connected
+                
+                mockWebSocketProvider.send(data)
+            }
+        }
+        delegate.receiveEventCallback = { data, error in
+            if error == nil {
+                receiveDataSend.fulfill()
+            }
+        }
+        
+        client.setDelegate(delegate, callbackQueue: DispatchQueue.global())
+        
+        client.startTranscriptionWSS(request)
+        //add waiter to open web socket and make sure its connected before sending data.
+        
+        wait(for: [receiveDataSend], timeout: 10)
+    }
+}
+
+class MockAWSTranscribeStreamingClientDelegate: NSObject, AWSTranscribeStreamingClientDelegate {
+    func connectionStatusDidChange(_ connectionStatus: AWSTranscribeStreamingClientConnectionStatus, withError error: Error?) {
+        
+    }
+    
+    func didReceiveEvent(_ event: AWSTranscribeStreamingTranscriptResultStream?, decodingError: Error?) {
+        
+    }
+}
+
+class MockWebSocketProvider: NSObject, AWSTranscribeStreamingWebSocketProvider{
+    
+    var webSocket: AWSTranscribeStreamingWebSocketProvider
+    
+    var clientDelegate: AWSTranscribeStreamingClientDelegate
```

Comment:
make this `private(set)` to enforce read-onlyness outside this class.

Reply:
done

Reply:
can't. 
<img width="1116" alt="Screen Shot 2020-01-07 at 1 49 03 PM" src="https://user-images.githubusercontent.com/8533749/71932333-8664fa80-3154-11ea-86ed-a62f3e8d48f5.png">